apps/gnome-mobile/src/app/(app)/(tabs)/index.tsx

```diff
@@ -96,19 +129,64 @@ const MapScreen = () => {
     };
   }, []);
 
-  let text = "Waiting...";
-  if (errorMsg) {
-    text = errorMsg;
-  }
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      const calculateDistance = (lat1, lon1, markers) => {
+        const R = 6371; // km
+        const distances = markers.map((marker) => {
+          const dLat = ((marker.latitude - lat1) * Math.PI) / 180;
+          const dLon = ((marker.longitude - lon1) * Math.PI) / 180;
+          const lat1Rad = (lat1 * Math.PI) / 180;
+          const lat2Rad = (marker.latitude * Math.PI) / 180;
+
+          const a =
+            Math.sin(dLat / 2) * Math.sin(dLat / 2) +
+            Math.sin(dLon / 2) *
+              Math.sin(dLon / 2) *
+              Math.cos(lat1Rad) *
+              Math.cos(lat2Rad);
+          const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+          const d = R * c;
+
+          return d;
+        });
+
+        return distances;
+      };
```

Comment:
na mainie Maciek użył fajnej biblioteki która ten dystans obliczy za nas, będzie to lepsze niż ta matematyka wyżej :D